CHANGELOG.md

```diff
@@ -33,10 +33,14 @@ x.x.x Release notes (yyyy-MM-dd)
 * The return types of the `SyncError.clientResetInfo()` and
   `SyncError.deleteRealmUserInfo()` APIs have been changed. They now return
   `RLMSyncErrorActionToken`s or `SyncError.ActionToken`s instead of closures.
+* Swift `Object` comparison and hashing behavior now works the same way as
+  that of `RLMObject` (objects are now only considered equatable if their
+  model class defines a primary key).
```

Comment:
Yes, this should be changed. NSObjects are always `Equatable` according to Swift; the change now is that we don't break the contract for hash values anymore.